build.gradle

```diff
@@ -1,22 +1,22 @@
 plugins {
     // we can not use ext variables here
-    id "java"
+    id "application"
     id "com.diffplug.spotless" version "6.23.3"
```

Comment:
wäre das eine stelle, wo man evtl. auf die version verzichten kann? würde dann einfach immer das aktuell spotless genommen? letztlich machen wir das ja aktuell manuell so: wir schalten immer auf die aktuelle version um ... oder könnte man diese version auch über `ext{}` definieren und hier referenzieren?

Reply:
hier brauchen wir explizit eine Versionsnummer `(plugin dependency must include a version number for this source)`. `ext{}` kann bei den plugins nicht referenziert werden. aber wir könnten den + Operator nutzen: `id "com.diffplug.spotless" version "6.+"`.

Reply:
ist das ok so?

Reply:
ehrlich gesagt verstehe ich nicht, warum spotless hier eine versionsnummer erzwingt. eigentlich ist mir die komplett egal, ich will einfach die aktuelle version nutzen.

Reply:
> ehrlich gesagt verstehe ich nicht, warum spotless hier eine versionsnummer erzwingt.

da es kein core plugin ist, sondern ein community plugin. siehe [hier](https://docs.gradle.org/current/userguide/plugins.html#plugin_distribution). deshalb muss eine version angegeben werden. das ist eine limitierung seitens gradle.

> eigentlich ist mir die komplett egal, ich will einfach die aktuelle version nutzen.

dann können wir den + operator nutzen: `id "com.diffplug.spotless" version "+"` und haben immer die neuste version.

ich würde das noch ändern(?)

Reply:
boah. ja, dann mach das so.

Reply:
fertig 🙂